python/ray/tune/trial_executor.py

```diff
@@ -15,32 +16,53 @@
 logger = logging.getLogger(__name__)
 
 
+# Ideally we want to use @cache; but it's only available for python 3.9.
+# Caching is only helpful for no autoscaler case.
 @lru_cache()
-def _get_warning_threshold() -> float:
+def _get_cluster_resources_no_autoscaler() -> Dict:
+    return ray.cluster_resources()
+
+
+def _can_fulfill_no_autoscaler(trial: Trial) -> bool:
+    """Calculates if there is enough resources for a PENDING trial.
+
+    For no autoscaler case.
+    """
+    assert trial.status == Trial.PENDING
+    return trial.resources.cpu <= _get_cluster_resources_no_autoscaler().get(
+        "CPU", 0
+    ) and trial.resources.gpu <= _get_cluster_resources_no_autoscaler().get(
+        "GPU", 0)
```

Comment:
hmm, this ignores any custom resources (trial.resources.custom_resources); might not be a problem right now, but we may want to fix this later.

Reply:
Hmm, I see. I may need some context around extra/custom/extra custom resources.

Reply:
Read a bit more about them. Are these concepts going away with PlacementGroup?

Reply:
custom resources will not go away; extra resources will go away.